src/Transit/Internal/Peer.hs

```diff
@@ -218,34 +220,34 @@ sendRecord ep record = do
   -- format sz as a fixed 4 byte bytestring
   let payloadSize = toLazyByteString (word32BE (fromIntegral (BS.length record)))
   _ <- sendBuffer ep (toS payloadSize)
-  _ <- sendBuffer ep record
+  _ <- sendBuffer ep (toS record)
   return ()
 
--- | Given the record encryption key and a bytestream, chop
--- the bytestream into blocks of 4096 bytes, encrypt them and
--- send it to the given network endpoint. The length of the
--- encrypted record is sent first, encoded as a 4-byte
--- big-endian number. After that, the encrypted record itself
--- is sent. `sendRecords` returns the SHA256 hash of the unencrypted
--- file, which can be compared with the recipient's sha256 hash.
-sendRecords :: TCPEndpoint -> SecretBox.Key -> ByteString -> IO Text
-sendRecords ep key bytes = do
-  forM_ records (sendRecord ep)
-  return $ show (sha256sum blocks)
+encryptC :: Monad m => SecretBox.Key -> C.ConduitT ByteString ByteString m ()
+encryptC key = go Saltine.zero
+  where
+    go nonce = do
+      b <- C.await
+      case b of
+        Nothing -> return ()
+        Just chunk -> do
+          let ct = encrypt key nonce chunk
+              ctsz = toLazyByteString (word32BE (fromIntegral (BS.length ct)))
```

Comment:
and `cipherTextSize` here?

Reply:
Thanks, I will fix these and submit a new PR today.